edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/CloudProxy.cs

```diff
@@ -153,7 +153,7 @@ public async Task SendMessageAsync(IMessage inputMessage)
                 using (Metrics.TimeMessageSend(this.clientId, outputRoute))
                 {
                     Metrics.MessageProcessingLatency(this.clientId, inputMessage);
-                    await this.client.SendEventAsync(message);
+                    await this.client.SendEventAsync(message).TimeoutAfter(this.sdkWaitTimeSeconds, new EdgeHubCloudSDKException());
                     Events.SendMessage(this);
```

Comment:
I know that we talked about this and I said it was ok, but now looking at it I have bad feelings: it always creates an instance of the exception for every call, but the 99.999999% of the cases it will not be needed (basically it will be needed only for that very very rare error we are fixing). On the other hand sometimes we send several messages per second. So it seems an unnecessary overhead.

My first idea was to create a readonly instance (of the exception) and then reuse that. However an internet search showed that it may cause problems: https://stackoverflow.com/questions/49534361/throw-same-exception-instance-multiple-times

So I would go back to the action, which would create an exception dynamically but only when it is needed. That is a better known pattern anyway (probably for a reason) Sorry, that I could not come up with a reasoning earlier, when we talked about it.

Reply:
That makes total sense Viktor. I had initially done that when you proposed it, so I'll revert my work and push it

Reply:
Why does this create an exception instance for every call? The instance will be created only if the callback is executed, right?

Reply:
This thread is outdated. I spoke to Viktor and changed the timeout extension to utilize an action instead. As Viktor said, that would create the exception dynamically instead of every time